plan/task_profile.go

```diff
@@ -241,59 +270,82 @@ func (sel *Selection) attach2TaskProfile(profiles ...taskProfile) taskProfile {
 	profile := profiles[0].copy()
 	switch t := profile.(type) {
 	case *copTaskProfile:
-		if t.addPlan2Index {
-			var indexSel, tableSel *Selection
-			if t.tablePlan != nil {
-				indexSel, tableSel = sel.splitSelectionByIndexColumns(t.indexPlan.Schema())
-			} else {
-				indexSel = sel.Copy().(*Selection)
-			}
-			if indexSel != nil {
-				indexSel.SetChildren(t.indexPlan)
-				t.indexPlan = indexSel
-				// TODO: Update t.cnt.
-				t.cst += float64(t.cnt) * cpuFactor
-			}
-			if tableSel != nil {
-				t.finishIndexPlan()
-				tableSel.SetChildren(t.tablePlan)
-				t.tablePlan = tableSel
-				t.cst += float64(t.cnt) * cpuFactor
+		var indexConds, tableConds []expression.Expression
+		conds := sel.pushDownConditions
+		if t.indexPlanFinished {
+			if ts, ok := t.tablePlan.(*PhysicalTableScan); ok {
+				conds = ts.filterCondition
 			}
+		} else if is, ok := t.indexPlan.(*PhysicalIndexScan); ok {
+			conds = is.filterCondition
+		}
+		if t.tablePlan != nil && !t.indexPlanFinished {
```

Comment:
Think the first `if` use single condition is better, like
```
if t.indexPlanfinished {
...
} else if t.tablePlan != nil {
...
} else {
...
}
```
This way, every condition is checked only once.